#!/usr/bin/env python
from dynamic_reconfigure.parameter_generator_catkin import *
PACKAGE = "motion_controller"


gen = ParameterGenerator()

gen.add("r_start", int_t, 0, "row start", 131, 0, 240)
gen.add("r_end", int_t, 0, "row end", 195, 0, 240)
# r_start < r_end
gen.add("c_start", int_t, 0, "col start", 16, 0, 160)
# c_end = 320 - c_start
gen.add("mask_r_start", int_t, 0, "mask's row start", 35, 0, 240)
# mask_r_start < r_end - r_start
gen.add("mask_c_start", int_t, 0, "mask's col start", 100, 0, 160)
# mask_c_start < (c_end - c_start) / 2.0
gen.add("judge_line", int_t, 0, "where calculate error", 20, 0, 240)
# judge_line < r_end - r_start
# gen.add("threshold", int_t, 0, "threshold", 160, 0, 255)
gen.add("Hough_threshold", int_t, 0, "HoughLines threshold", 40, 0, 100)
gen.add("linear_velocity", double_t, 0, "twist.linear.x", 0.2, 0, 0.3)
gen.add("rho_thr", double_t, 0, "rho threshold", 5, 0, 20)
gen.add("theta_thr", double_t, 0, "theta threshold", 3, 0, 20)
gen.add("kp", double_t, 0, "PID's P", 0.03, 0, 0.1)
gen.add("kd", double_t, 0, "PID's D", 0.015, 0, 0.1)
gen.add("v_black_up", int_t, 0, "black V upperbound", 100, 50, 150)
gen.add("motor_status", bool_t, 0, "Motor move or not", False)

exit(gen.generate(PACKAGE, "motion_controller", "line"))